org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/State.java

```diff
@@ -62,14 +70,12 @@ public class State {
 // keyed by the project relative path of the type (i.e. "src1/p1/p2/A.java"), value is a ReferenceCollection or an AdditionalTypeCollection
 Map<String, ReferenceCollection> references;
 // keyed by qualified type name "p1/p2/A", value is a map that maps a release to the project relative path which defines this type "src1/p1/p2/A.java"
-public Map<String, Map<Integer,String>> typeLocators;
+public TypeLocator typeLocators;
```

Comment:
Thanks for going the extra mile.

As a final polish I suggest to rename the class to `TypeLocators` (matching also the variable name), to indicate that one instance of that class manages many `String typeLocator` instances.

In that some round also the comment above the field should be updated / perhaps moved to the new class.